### PR TITLE
feat: #1369 The Web Components Config API should store config by appId

### PR DESCRIPTION
**Pull request by phmngocnghia**
# Pull request checklist

## Does this close any currently open issues?

fixes: #1369 

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
old route:
`
webComponentsConfig.get('/:customerId', webComponentsConfigGetByIdHandler)
webComponentsConfig.put('/', webComponentsConfigPutHandler)
webComponentsConfig.patch('/:customerId', webComponentsConfigPatchHandler)
webComponentsConfig.delete('/:customerId', webComponentsConfigDeleteHandler)
`

Issue Number: #1369 

## What is the new behavior?
Add `appId` to all CRUD routes

`
webComponentsConfig.post('/:customerId/:appId', webComponentsConfigPostHandler)
webComponentsConfig.get('/:customerId/:appId', webComponentsConfigGetByIdHandler)
webComponentsConfig.delete('/:customerId/:appId', webComponentsConfigDeleteHandler)
webComponentsConfig.patch('/:customerId/:appId', webComponentsConfigPatchHandler)
webComponentsConfig.put('/:customerId/:appId', webComponentsConfigPutHandler)
`

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

**Will need to add global secondary index key to `appId` field: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/getting-started-step-6.html**

### CODE REVIEW

File: packages/web-components-config-server/src/schemas/schema.ts [hunk 1]
```diff
@@ -7,6 +7,9 @@ export class WebComponentConfig {
   @hashKey()
   customerId: string
 
+  @hashKey()
```
Comment (vuhuucuong) [inline packages/web-components-config-server/src/schemas/schema.ts:10]: Something's wrong with this, in the issue's description, the returned data should be
```
{
   "DXX":{
      "appId1":{
         "config":{
            "appointmentLength":30,
            "appointmentTimeGap":60,
            "appointmentTypes":[
               {
                  "id":"id1",
                  "value":"value1"
               }
            ],
            "daysOfWeek":[
               "2",
               "1",
               "4",
               "5",
               "3"
            ],
            "negotiatorIds":[
               "AAAN",
               "AANW",
               "AAAW",
               "AAPW",
               "LXJ",
               "GXR",
               "FRED",
               "IEOQ",
               "MBW",
               "HOPH",
               "PMH",
               "MLB"
            ]
         }
      },
      "appId2":{
         "config":{

         }
      }
   }
}
```
But in your PR, the data will be like this
```
{
   "appId":"appId1",
   "appointmentLength":30,
   "appointmentTimeGap":60,
   "appointmentTypes":[
      {
         "id":"id1",
         "value":"value1"
      }
   ],
   "customerId":"DXX",
   "daysOfWeek":[
      "2",
      "1",
      "4",
      "5",
      "3"
   ],
   "negotiatorIds":[
      "AAAN",
      "AANW",
      "AAAW",
      "AAPW",
      "LXJ",
      "GXR",
      "FRED",
      "IEOQ",
      "MBW",
      "HOPH",
      "PMH",
      "MLB"
   ]
}
```

Comment (phmngocnghia) [reply packages/web-components-config-server/src/schemas/schema.ts:10]: It's WIP, I just changed the title to test the PR-lint

Comment (phmngocnghia) [reply packages/web-components-config-server/src/schemas/schema.ts:10]: The return schema type is an e.g. about how document structure look like, not to enforce return schema

File: packages/web-components-config-server/src/routes/web-components-config/web-components-config.ts [hunk 2]
```diff
@@ -67,9 +54,26 @@ export const webComponentsConfigDeleteHandler = async (req: AppRequest, res: App
...
+      data: { ...req.body, customerId: req.params.customerId, appId: req.params.appId },
+      traceId: req.traceId,
+    }
+    const { data } = params
+
+    validateCreate(data)
+    const result = await createConfig(params)
+    return res.send(result)
+  } catch (err) {
+    // logger.error('webComponentsConfig.create', { traceId: req.traceId, error: JSON.stringify(err) })
```
Comment (tanphamhaiduong) [inline packages/web-components-config-server/src/routes/web-components-config/web-components-config.ts:69]: Does it temporary ?

File: packages/web-components-config-server/src/routes/web-components-config/web-components-config.ts [hunk 3]
```diff
@@ -67,9 +54,26 @@ export const webComponentsConfigDeleteHandler = async (req: AppRequest, res: App
...
 }
 
-webComponentsConfig.get('/:customerId', webComponentsConfigGetByIdHandler)
-webComponentsConfig.put('/', webComponentsConfigPutHandler)
-webComponentsConfig.patch('/:customerId', webComponentsConfigPatchHandler)
-webComponentsConfig.delete('/:customerId', webComponentsConfigDeleteHandler)
+export const webComponentsConfigPostHandler = async (req: AppRequest, res: AppResponse) => {
+  try {
+    const params = {
+      data: { ...req.body, customerId: req.params.customerId, appId: req.params.appId },
```
Comment (tanphamhaiduong) [inline packages/web-components-config-server/src/routes/web-components-config/web-components-config.ts:60]: shoud we use opt chain to check null or undefined ?

Comment (phmngocnghia) [reply packages/web-components-config-server/src/routes/web-components-config/web-components-config.ts:92]: It's params from express route matching. Pretty sure about value of these field are always something not null

File: packages/web-components-config-server/src/routes/web-components-config/api.ts [hunk 4]
```diff
@@ -31,13 +39,26 @@ export const createConfig = async ({ traceId, data }: CreateParams): Promise<Web
...
+    await logger.error('Patch config failed', { traceId, error: stringifyError(error) })
+    throw error
+  }
+}
+
+export const putConfig = async ({ traceId, data }): Promise<WebComponentConfig> => {
+  try {
+    logger.info('Updating config...', { traceId, data })
+    const itemToUpdate = generateSchemaItem(data)
+    const result = await dynamoDBMapper.put(itemToUpdate)
```
Comment (vuhuucuong) [inline packages/web-components-config-server/src/routes/web-components-config/api.ts:61]: just wonder why we need both PUT, and PATCH to update ?

Comment (tanphamhaiduong) [reply packages/web-components-config-server/src/routes/web-components-config/api.ts:61]: @phmngocnghia please fix to the next PR

**Pull request merged**